lib/avalon/batch/entry.rb

```diff
@@ -18,126 +18,150 @@
+                      "(Must be one or the other)") unless files.empty?
+        else
           # Ensure files are listed
           @errors.add(:content, "No files listed") if files.empty?
-          # Replace collection error if collection not found
-          if media_object.collection.nil?
-            @errors.messages[:collection] = ["Collection not found: #{@fields[:collection].first}"]
-            @errors.messages.delete(:governing_policy)
-          end
-
-          return @errors.messages.empty?
+        end
+        # Replace collection error if collection not found
+        if media_object.collection.nil?
+          @errors.messages[:collection] = ["Collection not found: #{@fields[:collection].first}"]
+          @errors.messages.delete(:governing_policy)
         end
 
-        def file_valid?(file_spec)
-          valid = true
-          # Check date_digitized for valid format
-          if file_spec[:date_digitized].present?
-            begin
-              DateTime.parse(file_spec[:date_digitized])
-            rescue ArgumentError
-              @errors.add(:date_digitized, "Invalid date_digitized: #{file_spec[:date_digitized]}. Recommended format: yyyy-mm-dd.")
-              valid = false
-            end
-          end
-          # Check file offsets for valid format
-          if file_spec[:offset].present? && !Avalon::Batch::Entry.offset_valid?(file_spec[:offset])
-            @errors.add(:offset, "Invalid offset: #{file_spec[:offset]}")
+        return @errors.messages.empty?
+      end
+
+      def file_valid?(file_spec)
+        valid = true
+        # Check date_digitized for valid format
+        if file_spec[:date_digitized].present?
+          begin
+            DateTime.parse(file_spec[:date_digitized])
+          rescue ArgumentError
+            @errors.add(:date_digitized, "Invalid date_digitized: #{file_spec[:date_digitized]}. Recommended format: yyyy-mm-dd.")
             valid = false
           end
-          # Ensure listed files exist
-          if File.file?(file_spec[:file]) && self.class.derivativePaths(file_spec[:file]).present?
-            @errors.add(:content, "Both original and derivative files found")
-            valid = false
-          elsif File.file?(file_spec[:file])
+        end
+        # Check file offsets for valid format
+        if file_spec[:offset].present? && !Avalon::Batch::Entry.offset_valid?(file_spec[:offset])
+          @errors.add(:offset, "Invalid offset: #{file_spec[:offset]}")
+          valid = false
+        end
+        # Ensure listed files exist
+        if File.file?(file_spec[:file]) && self.class.derivativePaths(file_spec[:file]).present?
+          @errors.add(:content, "Both original and derivative files found")
+          valid = false
+        elsif File.file?(file_spec[:file])
+          #Do nothing.
+        else
+          if self.class.derivativePaths(file_spec[:file]).present? && file_spec[:skip_transcoding]
```

Comment:
Maybe this nested if structure would be more readable as a switch/case? It would ease confusion between the inner and outer ifs